src/screens/PinFlow/CurrentPinScreen.tsx

```diff
@@ -59,11 +120,30 @@ export class CurrentPinScreen extends PureComponent<Props, State> {
           <PinInput value={this.state.pin} onTextChange={this.updatePin} />
           <Text style={styles.errorText}>{error}</Text>
         </View>
+        {this.isTimeCounterVisible() && (
+          <TimeCounter
+            navigation={this.props.navigation}
+            onCountFinish={this.onCountFinish}
+            isVisible={this.isTimeCounterVisible()}
+            timestamp={this.props.timeCounter.timestamp}
+          />
+        )}
       </ScreenTemplate>
     );
   }
 }
 
+const mapStateToProps = (state: ApplicationState) => ({
+  timeCounter: state.timeCounter,
+});
+
+const mapDispatchToProps = {
```

Comment:
what is there a reason that we need redux for this? it seems that the state can be local

Reply:
Redux is needed because state should be saved after closing the app